### PR TITLE
implementation of LanCommunicator Class

### PR DESCRIPTION
**Pull request by sunny117**
fixes convict-git#50

### CODE REVIEW

File: src/networking/LanCommunicator.java [hunk 1]
```diff
@@ -1,27 +1,171 @@
...
+	        sendQueueListenerWorker = new Thread(sendQueueListener);
+	        sendQueueListenerWorker.start();
+	        
+	        logger.log(ModuleID.NETWORKING, logLevel.INFO, "sendQueueListener thread started");
+	        
+	        /** 
+	         * The listener that will be listening on the network and that receives the packet sent by the sendQueueListener
+	         * This listener will distingush between processing module message and content module's message and push into their respective queues 
+	         */
+	        socketListener = new SocketListener(processingReceiveQueue, contentReceiveQueue, portNumber);
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: SocketListener takes first argument as portnumber ,change here

File: src/networking/LanCommunicator.java [hunk 2]
```diff
@@ -1,27 +1,171 @@
...
+	        
+	        /** 
+	         * This listener will be listening on the receive queue which is for the content modules message
+	         * It will send the message which is pushed by network listener through the content module handler 
+	         */
+	        contentReceiveQueueListener = new ReceiveQueueListener(contentReceiveQueue, handlerMap);
+	        contentReceiveQueueListenerWorker = new Thread(contentReceiveQueueListener);
+	        contentReceiveQueueListenerWorker.start();
+	        
+	        logger.log(ModuleID.NETWORKING, logLevel.INFO, "contentReceiveQueueListener thread started");
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: change loglevel to LogLevel

File: src/networking/LanCommunicator.java [hunk 3]
```diff
@@ -1,27 +1,171 @@
...
+	
+	        logger.log(ModuleID.NETWORKING, logLevel.INFO, "socketListener thread started");
+	        
+	        /** 
+	         * This listener will be listening on the receive queue which is for the processing modules message
+	         * It will send the message which is pushed by network listener through the processing module handler 
+	         */
+	        processingReceiveQueueListener = new ReceiveQueueListener(processingReceiveQueue, handlerMap);
+	        processingReceiveQueueListenerWorker = new Thread();
+	        processingReceiveQueueListenerWorker.start();
```
Comment (sravand123) [inline src/networking/LanCommunicator.java:130]: Declare these worker threads  before using it

File: src/networking/LanCommunicator.java [hunk 4]
```diff
@@ -1,27 +1,171 @@
...
+    HashMap<String, INotificationHandler> handlerMap;
+    
+    /** Port Number that we will be listening on the network */
+    int portNumber;
+    
+    /** Boolean check variable for listeners to use */
+    static boolean isRunning = false;
+    
+    /** Logger instance for logging errors and activities */
+    ILogger logger;
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: get logger from LoggerFactory
```Java  
     ILogger logger =  LoggerFactory.getLoggerInstance();

File: src/networking/LanCommunicator.java [hunk 5]
```diff
@@ -1,27 +1,171 @@
...
 
+    /**
+     * This method takes identifier and handler
+     * and maps them so that they can be passed to the ReceiveQueueListeners
+     * 
+     * @param identifier
+     * @param handler
+     */
+    @Override
     public void subscribeForNotifications(String identifier, INotificationHandler handler){
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: Please remove this  implementation

**Review by sravand123 — CHANGES_REQUESTED** [review]

File: src/networking/LanCommunicator.java [hunk 6]
```diff
@@ -1,27 +1,171 @@
...
-class LanCommunicator implements ICommunicator{
+import java.util.HashMap;
+import networking.queueManagement.*;
+import networking.utility.*;
+import infrastructure.validation.logger.*;
 
+public class LanCommunicator implements ICommunicator{
+
+    /** SendQueueListener that will be listening on the send Queue */
+    SendQueueListener sendQueueListener;
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: Make all the fields as private

**Review by sravand123 — CHANGES_REQUESTED** [review]

File: src/networking/LanCommunicator.java [hunk 7]
```diff
@@ -1,27 +1,171 @@
...
+    IQueue<IncomingPacket> contentReceiveQueue;
+
+    /** Hashmap contains the handlers for the respective modules processing and content */
+    HashMap<String, INotificationHandler> handlerMap;
+    
+    /** Port Number that we will be listening on the network */
+    int portNumber;
+    
+    /** Boolean check variable for listeners to use */
+    static boolean isRunning = false;
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: use getter and setter for this variable

**Review by sravand123 — CHANGES_REQUESTED** [review]

**Comment by sunny117** [comment]
Updated the requested changes

File: src/networking/LanCommunicator.java [hunk 8]
```diff
@@ -1,27 +1,186 @@
...
+    ILogger logger = LoggerFactory.getLoggerInstance();
+
+    /**
+     * constructor that takes port and also initializes a hashmap
+     * 
+     * @param port that we will be passing to the socketListener
+     */
+    public LanCommunicator(int port) {
+    	portNumber = port;
+        handlerMap = new java.util.HashMap<>();
```
Comment (sravand123) [inline src/networking/LanCommunicator.java]: change java.util.HashMap, it is redundant

**Review by sravand123 — CHANGES_REQUESTED** [review]

File: src/networking/LanCommunicator.java (hunk 8 above)
Comment (sunny117) [reply src/networking/LanCommunicator.java]: fixed the requested changes

**Review by sravand123 — APPROVED** [review]
Everything looks good!! Excellent work

**Pull request merged**